core/src/cluster_slot_state_verifier.rs

```diff
@@ -33,147 +45,185 @@ impl SlotStateUpdate {
     fn to_handler(&self) -> SlotStateHandler {
         match self {
             SlotStateUpdate::Dead => on_dead_slot,
-            SlotStateUpdate::Frozen => on_frozen_slot,
-            SlotStateUpdate::DuplicateConfirmed => on_cluster_update,
-            SlotStateUpdate::Duplicate => on_cluster_update,
+            SlotStateUpdate::BankFrozen => on_frozen_slot,
+            SlotStateUpdate::DuplicateConfirmed => on_duplicate_confirmed,
+            SlotStateUpdate::Duplicate => on_duplicate,
         }
     }
 }
 
-fn on_dead_slot(
+fn check_duplicate_confirmed_hash_against_frozen_hash(
+    state_changes: &mut Vec<ResultingStateChange>,
     slot: Slot,
-    bank_frozen_hash: &Hash,
-    cluster_duplicate_confirmed_hash: Option<&Hash>,
-    _is_slot_duplicate: bool,
+    cluster_duplicate_confirmed_hash: Option<Hash>,
```

Comment:
should we just pass the args in here? seems like it's all the fields anyway

Reply:
Not after Justin's latest suggestion: https://github.com/solana-labs/solana/pull/18648#discussion_r670763448